Added initial code to generate binary and package via GooGet

james-bebbington opened the pull request.

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+# set GOOARCH based on ARCH
+ifeq ($(ARCH),x86_64)
```
these 3 expressions can be rewritten with `else`:
```
ifeq ($(ARCH),x86_64)
  GOARCH=amd64
else ifeq ($(ARCH),x86)
  GOARCH=386
else
  $(error "ARCH must be set to one of: x86, x86_64")
endif
```

james-bebbington replied:
Much nicer, thnx. Note as much as it looks weird to not indent the code inside the conditions, I did find I had to not indent the variable setters otherwise I would get an error.

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+# set GOOARCH based on ARCH
```
typo: `GOARCH`

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+$(error "ARCH must be set to one of: x86, x86_64")
+endif
+
+# use "0.0.1" as default version if not supplied
```
why `0.0.1` specifically? should it be some invalid version instead, e.g. `0.0.0`, or a git hash?
You can probably use something like `git describe --tags --dirty` for automatic versioning

james-bebbington replied:
I still need to finalize how exactly versioning will work. Googet only supports versions using three dot notation, so it's a bit limited. This will become clearer once I get more understanding of how the package upload process works.

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+
+.PHONY: build
+build:
+	GOOS=windows go build -o ./bin/otelopscol_windows_$(GOARCH).exe ./cmd/otelopscol
```
Do you plan to eventually make this command support other OSes (Linux / Mac)?

james-bebbington replied:
If we go with a Makefile per build image (similar to how the existing Stackdriver agents are built), then this command will probably only support Windows. We may find, however, that we only need the one image to build the Collector, in which case we can add extra parameters to this command as needed.

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+
+.PHONY: pack-googet
+pack-googet:
+	# googet doesn't support variable replacement or comamand line args so just use sed
```
typo: `command`

nilebox commented on .build/Makefile:
```diff
@@ -0,0 +1,30 @@
+.PHONY: pack-googet
+pack-googet:
+	# googet doesn't support variable replacement or comamand line args so just use sed
+	sed 's/{{$$VERSION}}/$(VERSION)/g;s/{{$$ARCH}}/$(ARCH)/g;s/{{$$GOARCH}}/$(GOARCH)/g' \
```
`sed` works, but `envsubst` seems to be more common / safer way to substitute variables (if your base Docker image supports it).
https://unix.stackexchange.com/a/294400

james-bebbington replied:
👍

I did have to add `gettext-base` to the build docker image annoyingly, but I think that's fine.

james-bebbington commented:
@nilebox Thnx for the quick review! I addressed all your comments, and I'm going to go ahead and merge this even though it's not perfect yet as I want to have some base code in the repo :)
